MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp

```diff
@@ -193,345 +207,475 @@ void EnggDiffFittingPresenter::fittingFinished() {
-
-    m_view->setFittingRunNumVec(runnoDirVector);
-
-    auto multiRunMode = m_view->getFittingMultiRunMode();
-    auto singleRunMode = m_view->getFittingSingleRunMode();
-    // if not run mode or bank mode: to avoid recreating widgets
-    if (!multiRunMode && !singleRunMode) {
-
-      // add bank to the combo-box
-      setBankItems();
-      // set the bank widget according to selected bank file
-      setDefaultBank(splitBaseName, strFocusedFile);
-      runNoVec.clear();
-      runNoVec.push_back(splitBaseName[1]);
-
-      // Skips this step if it is multiple run because widget already
-      // updated
-      setRunNoItems(runNoVec, false);
-    }
+  const std::string workingDirectory = pocoFullFilePath.parent().toString();
+
+  // Find all files which match this baseFilenamePrefix -
+  // like a poor mans regular expression for files
+  if (!findFilePathFromBaseName(workingDirectory, baseFilenamePrefix,
+                                foundFullFilePaths)) {
+    // I can't see this ever being thrown if the user is browsing to files but
+    // better to be safe and give an informative message
+    throw std::runtime_error("Could not find any files matching the generated"
+                             " pattern: " +
+                             baseFilenamePrefix);
   }
+
+  // Store the run number as found
+  std::vector<std::string> runNoVec;
+  runNoVec.push_back(splitBaseName[1]);
+
+  return runNoVec;
 }
 
-void EnggDiffFittingPresenter::processMultiRun(
-    const std::string strFocusedFile,
-    std::vector<std::string> &runnoDirVector) {
+/**
+  * Processes a multi run input to the interface
+  * such as '12345-12350' by splitting it into '12345' to '12350'
+  * then calling enableMultiRun
+  * @param userInput The user input from the view
+  *
+  * @return List of found full file paths for the files specified
+  */
+std::vector<std::string>
+EnggDiffFittingPresenter::processMultiRun(const std::string userInput) {
+
+  // Split user input into the first and last run number
   std::vector<std::string> firstLastRunNoVec;
-  boost::split(firstLastRunNoVec, strFocusedFile, boost::is_any_of("-"));
+  boost::split(firstLastRunNoVec, userInput, boost::is_any_of("-"));
+
+  // Then store them in their own strings
   std::string firstRun;
   std::string lastRun;
   if (!firstLastRunNoVec.empty()) {
     firstRun = firstLastRunNoVec[0];
     lastRun = firstLastRunNoVec[1];
 
     m_view->setFittingMultiRunMode(true);
-    enableMultiRun(firstRun, lastRun, runnoDirVector);
   }
+  return enableMultiRun(firstRun, lastRun);
 }
 
-void EnggDiffFittingPresenter::processSingleRun(
-    const std::string &focusDir, const std::string &strFocusedFile,
-    std::vector<std::string> &runnoDirVector,
+std::vector<std::string> EnggDiffFittingPresenter::processSingleRun(
+    const std::string &userInputBasename,
     const std::vector<std::string> &splitBaseName) {
 
+  std::string focusDir = m_view->focusingDir();
```

Comment:
As a side note this is also part of #17433 as we don't check the user config for their directories